cpp/foxglove/include/foxglove/context.hpp

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <cstdint>
+#include <memory>
+#include <optional>
+#include <string>
+
+struct foxglove_context;
+
+namespace foxglove {
+
+typedef foxglove_context ContextInner;
+
+class Context final {
+  friend class McapWriterOptions;
+  friend class Channel;
+  friend class WebSocketServerOptions;
```

Comment:
I was going to say having to use `friend` seems like something to generally avoid, but I see the google c++ style guide allows it "within reason" https://google.github.io/styleguide/cppguide.html#Friends

A real 🤯 -- "If they’re used properly, they _enhance_ encapsulation" https://isocpp.org/wiki/faq/friends#friends-and-encap

Reply:
I was using it as an alternative to pub(crate) which we don't have in C++